src/libmongoc/tests/test-mongoc-cluster.c

```diff
@@ -1813,12 +1811,114 @@ _test_cmd_on_unknown_serverid (bool pooled)
 }
 
 void
-test_cmd_on_unknown_serverid (void)
+test_cmd_on_unknown_serverid_pooled (void)
 {
-   _test_cmd_on_unknown_serverid (false /* pooled */);
    _test_cmd_on_unknown_serverid (true /* pooled */);
 }
 
+void
+test_cmd_on_unknown_serverid_single (void)
+{
+   _test_cmd_on_unknown_serverid (false /* pooled */);
+}
+
+
+/* Test that server streams are invalidated as expected. */
+static void
+test_cluster_stream_invalidation_single (void)
+{
+   mongoc_client_t *client;
+   mongoc_server_description_t *sd;
+   bson_error_t error;
+   mongoc_server_stream_t *stream;
+
+   client = test_framework_new_default_client ();
+   /* Select a server to start monitoring. */
+   sd = mongoc_client_select_server (
+      client, true /* for writes */, NULL /* read prefs */, &error);
+   ASSERT_OR_PRINT (sd, error);
+
+   /* Test "clearing the pool". This should invalidate existing server streams.
+    */
+   stream = mongoc_cluster_stream_for_writes (
+      &client->cluster, NULL /* session */, NULL /* reply */, &error);
+   ASSERT_OR_PRINT (stream, error);
+   BSON_ASSERT (mongoc_cluster_stream_valid (&client->cluster, stream));
+   _mongoc_topology_clear_connection_pool (client->topology,
+                                           mongoc_server_description_id (sd));
+   BSON_ASSERT (!mongoc_cluster_stream_valid (&client->cluster, stream));
+   mongoc_server_stream_cleanup (stream);
+
+   /* Test closing the connection. This should invalidate existing server
+    * streams. */
+   stream = mongoc_cluster_stream_for_writes (
+      &client->cluster, NULL /* session */, NULL /* reply */, &error);
+   ASSERT_OR_PRINT (stream, error);
+   BSON_ASSERT (mongoc_cluster_stream_valid (&client->cluster, stream));
+   mongoc_cluster_disconnect_node (&client->cluster, sd->id);
+   BSON_ASSERT (!mongoc_cluster_stream_valid (&client->cluster, stream));
+   mongoc_server_stream_cleanup (stream);
+
+   /* Test that a new stream is considered valid. */
+   stream = mongoc_cluster_stream_for_writes (
+      &client->cluster, NULL /* session */, NULL /* reply */, &error);
+   ASSERT_OR_PRINT (stream, error);
+   BSON_ASSERT (mongoc_cluster_stream_valid (&client->cluster, stream));
+   mongoc_server_stream_cleanup (stream);
+
+   mongoc_server_description_destroy (sd);
+   mongoc_client_destroy (client);
+}
+
+/* Test that server streams are invalidated as expected. */
+static void
+test_cluster_stream_invalidation_pooled (void)
+{
+   mongoc_client_pool_t *pool;
+   mongoc_client_t *client;
+   mongoc_server_description_t *sd;
+   bson_error_t error;
+   mongoc_server_stream_t *stream;
+
+   pool = test_framework_new_default_client_pool ();
```

Comment:
Unlike my previous comment in `test-mongoc-client.c`, you _are_ using `test_framework_new_default_client_pool` instead of `test_framework_new_default_client` here. So I assume the other test might have had a mistake.

Reply:
Yes, the other test has been updated.